### DISCUSSION TITLE
backups

### DISCUSSION THREAD

**Discussion by Xel-Naha** [bpg/terraform-provider-proxmox#867] [Q&A]
hi all,

I currently use this provider for building vms and containers and one of things I still need to do manually (sort of) is adding containers and or vms which have been created into backup jobs.

are there any plans to make this part of the terraform provider as an option? I would very much love it if we could add a "backup" setting for vms and ct or even the whole cluster.

maybe a good idea? whats your thoughts?

**Comment by bpg-dev** [comment]
Hi @Xel-Naha!
That's definitely a good idea! We have a similar request #406 (without much of details though)
I just want to point out that terraform is not a good fit for executing a single backup job like "Backup Now" for a VM:
 
<img width="772" alt="Screenshot 2024-01-09 at 9 31 30 PM" src="https://github.com/bpg/terraform-provider-proxmox/assets/627562/d0f1b9f0-c45c-4e0d-89f3-00130a43db6a">

Bun managing backup jobs on cluster level is something that we can do:
<img width="846" alt="Screenshot 2024-01-09 at 9 32 45 PM" src="https://github.com/bpg/terraform-provider-proxmox/assets/627562/aab567f8-26d7-40b8-84b6-debb6a35081d">

**Comment by Xel-Naha** [comment]
hi @bpg 

I would agree with you that it might not be the best on an individual level. At the moment we organise it on a datacentre level and not on an individual level.

I would envision that we can set backup flag on a container/vm level and this gets reflected on cluster level. Happy to help testing if required as I don't write go, but happily help out testing.

**Comment by Wamarcaiceloff** [comment]
Привет @bpg есть ли какой то роадмап когда ожидать этот функционал? Мне очень нравится этот провайдер, но крайне не хватает этого функционала, я планировал дописать модуль сам, но увидел эту тему. Возможно у тебя уже есть наработки?

**Reply by bpg-dev** [reply]
Привет @Wamarcaiceloff 👋🏼 

Никаких наработок в этом направлении нет. Я бы посоветовал открыть новый enhancement тикет и более детально описать желаемый функционал, хорошо бы с притмерами terraform configs. Тогда это будет легче спланировать, и возможно кто-нибудь поучаствует в написании кода / тестировании.
